packages/widgets/src/tabbar.ts

```diff
@@ -1339,9 +1339,15 @@ namespace TabBar {
      *
      * @returns A virtual element representing the tab icon.
      */
-    renderIcon(data: IRenderData<any>): VirtualElement {
+    renderIcon(data: IRenderData<any>): VirtualElement | VirtualElementPass {
+      const { title } = data;
       let className = this.createIconClass(data);
-      return h.div({ className }, data.title.iconLabel);
+
+      if (title.iconRenderer) {
+        return hpass('div', title.iconRenderer);
+      } else {
+        return h.div({className}, data.title.iconLabel);
```

Comment:
If the `className` is completely disregarded when `title.iconRenderer` is set, then maybe `this.createIconClass(data)` should not be called at all?

Reply:
You're right, this is inconsistent, and kind of a holdover from an earlier point in the PR. I thought about it for a bit, and what I think should be done is that the `className` and `label` should be passed to the hpass element as well, to be set on its parent element. There's a PR for it at #36